### PR TITLE
[CDAP-19286] Fail master environment initialization if cluster role does not exist

### PR DESCRIPTION
**Pull request by dli357**
We want to fail early if the environment is misconfigured.

### CODE REVIEW

File: cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java [hunk 1]
```diff
@@ -400,6 +404,18 @@ public void initialize(MasterEnvironmentContext context) throws IOException, Ill
...
 
+  private void validateClusterRole(String roleName) throws ApiException, IllegalArgumentException {
+    try {
+      rbacV1Api.readClusterRole(roleName, null);
+    } catch (ApiException e) {
+      if (e.getCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+        throw new IllegalArgumentException(String.format("ClusterRole '%s' does not exist. Ensure it is created and " +
+                                                           "assigned to the master service account.", roleName), e);
+      }
+      throw e;
```
Comment (nliu132) [inline cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java]: can you wrap this in another exception in order to include `e.getResponseBody()`? IIRC the message in ApiException is useless for debugging

**Comment by nliu132** [comment]
this is unrelated, but since you already have a PR open can you fix this typo to use namespace instead of serviceAccountName https://github.com/cdapio/cdap/blob/74ed6c65d501296d3935b3853a8ac60ab6967eb8/cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java#L1062-L1063

File: cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java (hunk 1 above)
Comment (dli357) [reply cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java]: Fixed

**Comment by dli357** [comment]
> this is unrelated, but since you already have a PR open can you fix this typo to use namespace instead of serviceAccountName
> 
> https://github.com/cdapio/cdap/blob/74ed6c65d501296d3935b3853a8ac60ab6967eb8/cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java#L1062-L1063

Fixed, thanks!

**Review by nliu132 — APPROVED** [review]

**Pull request merged**